cmd/flux/cmd/fmt.go

```diff
@@ -0,0 +1,65 @@
+package cmd
+
+import (
+	"fmt"
+	"io/ioutil"
+
+	"github.com/influxdata/flux/ast"
+	"github.com/influxdata/flux/ast/astutil"
+	"github.com/influxdata/flux/parser"
+	"github.com/spf13/cobra"
+)
+
+// fmtCmd represents the fmt command
+var fmtCmd = &cobra.Command{
+	Use:   "fmt",
+	Short: "Format a Flux script-file",
+	Long:  "Format a Flux script-file (flux fmt [-w] <script-file>)",
+	Args:  cobra.MinimumNArgs(1),
+	RunE:  format,
+}
+
+var writeResultToSource bool
+
+func init() {
+	rootCmd.AddCommand(fmtCmd)
+	fmtCmd.Flags().BoolVarP(&writeResultToSource, "write-result-to-source", "w", false, "write result to (source) file instead of stdout")
+}
+
+func format(cmd *cobra.Command, args []string) error {
+	script := args[0]
+
+	fromFile, err := ioutil.ReadFile(script)
+	if err != nil {
+		return err
+	}
+
+	pkg := parser.ParseSource(string(fromFile))
+	if ast.Check(pkg) > 0 {
+		return fmt.Errorf("unexpected error: %s , filename: %s", ast.GetError(pkg), script)
+	} else if len(pkg.Files) != 1 {
+		return fmt.Errorf("expected one file in the package, got %d", len(pkg.Files))
+	}
+
+	formatStr, err := astutil.Format(pkg.Files[0])
```

Comment:
@jsternberg Mentioned in chat that he has seen some odd behavior when going over the Go Rust boundary multiple times using JSON serialization. 

The way the `parser.ParseSource` and `astutil.Format` APIs are being used here we are definitely serializing the AST multiple times.



Digging into the details, my understanding is that this is the process based on the code as written:

parser.ParseSource: Go string ->  Rust AST -> JSON -> Go AST

astutil.Format: Go AST -> JSON -> Rust AST -> Go string

Notice we could skip all of the serialization steps if we just kept the Rust AST and called format on it directly.:

Go string -> Rust AST ->  Rust string -> Go string 


The low level APIs we could use are [libflux.Parse](https://github.com/influxdata/flux/blob/e673e37e7c4b03ef8b98b0721e3e2d862edb275d/libflux/go/libflux/parser.go#L122) and [libflux.Format](https://github.com/influxdata/flux/blob/e673e37e7c4b03ef8b98b0721e3e2d862edb275d/libflux/go/libflux/parser.go#L36)



Two takeaways from this observation:

1. If we switched to using the lower level APIs do the odd parse errors go away?
2. We should fix whatever issues are arising from the JSON serialization. Perhaps we can create a test case that calls parse.ParseSouce and then astutil.Format on our everything.flux file to validate nothing changes.

Reply:
Thanks for the suggestion, the low-level APIs did work. I have updated the PR with the changes. 
I will work on point 2 separately as it's not a blocker for the current task.